tests/ut/conftest.py

```diff
@@ -29,27 +29,39 @@ def reset_caches():
 
 @pytest.fixture
 def mock_cache(mocker):
-    return create_autospec(BaseCache())
+    # TODO: Is there need for a separate BaseCache[bytes] fixture?
+    return create_autospec(BaseCache[str]())
 
 
 @pytest.fixture
 def mock_base_cache():
     """Return BaseCache instance with unimplemented methods mocked out."""
+    # TODO: Is there need for a separate BaseCache[bytes] fixture?
     plugin = create_autospec(BasePlugin, instance=True)
-    cache = BaseCache(timeout=0.002, plugins=(plugin,))
+    cache = BaseCache[str](timeout=0.002, plugins=(plugin,))
     methods = ("_add", "_get", "_gets", "_set", "_multi_get", "_multi_set", "_delete",
                "_exists", "_increment", "_expire", "_clear", "_raw", "_close",
                "_redlock_release", "acquire_conn", "release_conn")
     with ExitStack() as stack:
         for f in methods:
             stack.enter_context(patch.object(cache, f, autospec=True))
         stack.enter_context(patch.object(cache, "_serializer", autospec=True))
+        stack.enter_context(patch.object(cache, "build_key", cache._str_build_key))
         yield cache
 
 
+@pytest.fixture
+def generic_base_cache():
+    # TODO: Is there need for a separate BaseCache[bytes] fixture?
+    return BaseCache[str]()
```

Comment:
Also, task for another PR. BaseCache should be ABC, which means it then won't be able to be instantiated directly. This should probably then just use the memory cache for testing or something..

Reply:
Sounds good.